addons/udes_stock/tests/test_stock_warehouse.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+from .common import BaseUDES
+
+
+class TestStockWarehouseModel(BaseUDES):
+    @classmethod
+    def setUpClass(cls):
+        super(TestStockWarehouseModel, cls).setUpClass()
+        cls.Warehouse = cls.env['stock.warehouse']
+        # Get current warehouse - should only be one!
+        cls.test_warehouse = cls.Warehouse.search([])
+
+    def test00_get_picking_types(self):
+        """ Test checks if the test picking types are present """
+        test_picking_types = [
+            'TEST_PUTAWAY',
+            'TEST_INTERNAL',
+            'TEST_PICK',
+            'TEST_GOODS_OUT',
+            'TEST_TRAILER_DISPATCH',
+            'TEST_GOODS_IN',
+        ]
+        self.assertTrue(
+            set(test_picking_types).issubset(self.test_warehouse.get_picking_types().mapped('name'))
+        )
+
+    def test00_get_picking_types(self):
+        """ Test checks if the test picking types and the normal pick types are present """
+        test_picking_types = [
+            'TEST_PUTAWAY',
+            'TEST_INTERNAL',
+            'TEST_PICK',
+            'TEST_GOODS_OUT',
+            'TEST_TRAILER_DISPATCH',
+            'TEST_GOODS_IN',
+            'Goods In',
+            'Putaway',
+            'Goods Out',
+            'Pick',
+            'Trailer Dispatch',
+            'Internal Transfer',
+        ]
+        self.assertTrue(
+            set(test_picking_types).issubset(self.test_warehouse.get_picking_types().mapped('name'))
+        )
+
+    def test02_get_default_picking_types_new_warehouse(self):
+        """ Create a new warehouse and look at the default picking types """
+        test_picking_types = ['Receipts', 'Internal Transfers', 'Delivery Orders']
+        test_company = self.create_company('test_company')
+        test_warehouse = self.Warehouse.search([('company_id', '=', test_company.id)])
```

Comment:
There is a lot of warehouse searching by company id in this code.
A convenience function or relational field could be created.

Reply:
Changed, warehouse being used is stored in a class variable at common.py